module-2/backend/week_one/Ash.md

```diff
@@ -0,0 +1,104 @@
+  Get= Read part of CRUD
+  Post= Create part of CRUD
+  Put= Update part of CRUD
+  Delete= Delete part of CRUD
+
+What is Sinatra?
+  A framework which allows you to deploy an app/site, simple but effective, not as robust as Rails but will be recommended in certain situations.
+What is MVC?
+  Model / View / Controllers
+
+Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+  To be uniform and to ensure we create the necessary components.
+  
+What types of variables are accessible in our view templates without explicitly passing them?
+  Instance
+  
+Given the following block of code, how would I pass an instance variable count with a value of 1 to my index.erb template?
+
+get '/horses' do
+  @something = 1
+end
+
+In the same code block, how would I pass a local variable name with a value of Mr. Ed to the view?
+  :locals => {:mr_ed => mr_ed}
+
+What's the purpose of ERB?
+  Allows you to utilize Ruby as a resource within your HTML
+
+Why do I need a development AND test database?
+  So that you can have a database that doesn't get manipulated when you're trying to modify data.
+
+What's responsive design?
+  website that would adjust to resizing with having to distort content
+
+What is CRUD and why is it important?
+  Create, read, update, delete,
+  Important because you have a fully functional website
+
+What does HTTP stand for?
+  hyper text transfer protocol
+
+What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+<% %> and <%= =%>, the second will display value while first just runs the code
+
+What's an ORM?
+Object Relational Mapping.
+
+What's the most commonly used ORM in ruby (Sinatra & Rails)?
+Active Record and SQL
+
+Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+
+  get "/restaurants" do
+    @restaurant = Restaurant.all
+    erb :"restaurants/index"
+  end
+
+  get "/restaurant/new" do
+    erb :"restaurant/new"
+  end
+
+  post "/restaurant" do
+    restaurant = Restaurant.new(params[:restaurant])
+    restaurant.save
+    redirect "/restaurant"
+  end
+
+  get "/restaurant/:id" do
+    @restaurant = Restaurant.find(params[:id].to_i)
+    erb :"restaurant/show"
+  end
+
+  get "/restaurant/:id/edit" do
+    @restaurant = Restaurant.find(params[:id].to_i)
+    erb :"restaurant/edit"
+  end
+
+  put "restaurant/:id" do |id|
+    Restaurant.update(id.to_i, params[:restaurant])
+    redirect "/restaurant/#{id}"
+  end
+
+  delete "restaurant/:id" do
+    Restaurant.destroy(params[:id].to_i)
+    redirect "/restaurant"
+  end
+
+What's a migration?
+  Creating a procedure to update your database
+
+When you create a migration, does it automatically modify your database?
+  No, you have to to seed afterwards

```

Comment:
You have to *migrate*. Seeding will put data into your database, but migrating will modify the database.